### DISCUSSION TITLE
How to run a build target only when packing

### DISCUSSION THREAD

**Discussion by reflectronic** [NuGet/Home#12694] [Q&A]
In our NuGet package, we'd like to include some content files that are downloaded from the internet. My first thought was to create an MSBuild target that downloads the file and adds it as a `Content` item. Something like:

```xml
<Target Name="DownloadContent">
   <DownloadFile SourceUrl="..."
                 DestinationFolder="$(BaseIntermediateOutputPath)">
      <Output TaskParameter="DownloadedFile" ItemName="DownloadedConent" />
   </DownloadFile>

   <Unzip SourceFiles="@(DownloadedContent)"
          DestinationFolder="$(BaseIntermediateOutputPath)content" />

   <ItemGroup>
     <Content Pack="true" Include="$(BaseIntermediateOutputPath)content\*.*" />
   </ItemGroup>
</Target>
```

However, this raises the question: when should this target be executed? I only want this target to run when packing—running this as part of regular builds would be disruptive and slow. And, while `pack` has [extensibility points](https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#advanced-extension-points-to-create-customized-package) for adding custom content, they are for TFM-specific content. The content I'm downloading is not TFM-specific; I want this target to run in the outer build. (There's no need to download and extract the same content for each target framework.)

I can't find or think of a documented way to do this. It would be pretty easy for me to hack something together, but I don't want to cause additional maintenance burden for our team. My naive idea would be to have something like: 
```xml
<Target Name="DownloadContent" Condition="'$(IsPacking)' == 'true'">
```
but I don't know enough about NuGet to know if this idea is tenable or palatable.

Thanks.

**Comment by paulomorgado** [comment]
You need to add the target to `TargetsForTfmSpecificBuildOutput` or `TargetsForTfmSpecificBuildOutput`.

Use https://aka.ms/MSBuildLog to debug your build.
